main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
> BTW, CTags did something odd (for bad reasons if you ask me) with \rs without following \n: it handed them as bare characters, not terminating the line. However, various tools do consider them line ends on GNU/Linux at least.

Indeed! But that's actually great for me because I can keep it this way and just check if what I get from gets() ends with CRLF and convert it to LF and keep CRs within the string without having to do anything.

Great bug! :-)